app/models/case/sar/internal_review.rb

```diff
@@ -6,6 +6,19 @@ class Case::SAR::InternalReview < Case::SAR::Standard
 
   attr_accessor :original_case_number
 
+  validate :validate_email
+  validate :validate_postal_address
+  validate :name
+  validate :validate_subject
+  validate :validate_third_party_relationship
+
+  after_initialize :remove_conditional_validators
```

Comment:
@willmcb, I haven't tried it yet, but this line seems risky to me ,  and in my thought, it shouldn't need to do this. The problem with valid_attributes is that  it didn't really invoke the validator defined on specific attributes  correctly which means the validator will be skipped ( actually I start to doubt this valid_attributes? every works as I dug this issue a little more).   But those skips shouldn't affect your new validators in the SAR IR classes.   we can have discuss about it

Reply:
even there is possible conflict ( which I feel it shouldn't,  you should be allowed to have multiple validators against one field, as long as one of them failed, an error should be raised. ),  I more prefer to redefine those validators in parent class level, rather do "removal" actions

Reply:
I tend to agree, I added it this way as I thought changing them in the base class may have more side effects. Thinking about it though I think this way of doing it actually causes an issue in that you remove all of the validators on a given attribute, so unless you redeclare all of them some may be missing. I think I will just move them to the parent. Happy to chat about it though.

Reply:
Have added these to the parent class and it appears to be working as expected.

Reply:
@willmcb, I have thought about another methods to do the partial validation, we can discuss tomorrow, it only need to add a small function on controller level,  it is safer and no need to changes anything in the parent class, let us catch-up tomorrow

Reply:
I managed to make it work by removing the call to `.attributes_valid?` on case completion and then just assigning the attributes before the in final `@case.valid?` check. With the new methods I was having issues with the linking validation as `.attributes_valid?` calls the link validation logic in the form object via interpolating the current step into a method name. In any case it's fine now as the logic is so simple in the steps that we can rely on the preexisting validations in the `Case::SAR::Standard`. Also allowed me to remove the form validator for the SAR IR which is good as it means less code :)

Reply:
I see, yes, for you case there is no need to do the partial validation actually as first step and second step are really simple,  I agree it would be simpler,  although this won't apply to offender SAR flow 

since I thought about the new way of doing partial validation , I am quite tempting to refactor the form and validators for Offender SAR and complaint Flow , as the new way most of them can be removed, the logic would become much easy even for understanding, I may find a chance to do this refactor in the future